### PR TITLE
Scrape constituency names

### PR DESCRIPTION
**Pull request by ondenman**
Scrape member constituency names.

Members are grouped by [district on the wikipedia source page](https://en.wikipedia.org/wiki/List_of_MPs_elected_in_the_Mongolian_legislative_election,_2016). The district name is only displayed in the upper most row of each district (in column[0]). When the district name is present, it is stored. When a row is without a district name (column[0] contains an ordinal number), the stored district name is used.

### CODE REVIEW

File: scraper.rb [hunk 1]
```diff
@@ -9,6 +9,7 @@
 require 'pry'
 
 class Khurai
+
```
Comment (chrismytton) [inline scraper.rb:14]: This change doesn't seem related to this pull request. I would also add that Rubocop would report this as an offence - "Extra empty line detected at class body beginning". Probably best to amend the commit to remove this change :)

File: scraper.rb [hunk 2]
```diff
@@ -88,9 +86,15 @@ def tr_without_district
...
-      party: 4,
+      party: 4
     }
   end
+
+  def constituency
+    cell_text = tds[0].text.strip.gsub("\n", ' — ')
+    @current_constituency = cell_text unless cell_text =~ /\d/
+    current_constituency
+  end
```
Comment (chrismytton) [inline scraper.rb]: I don't think you need the `current_constituency` attribute as it's only being used in this method. Perhaps something like this would be better?

``` ruby
  def constituency
    cell_text = tds.first.text.strip.gsub("\n", ' — ')
    cell_text unless cell_text =~ /\d/
  end
```

File: scraper.rb [hunk 3]
```diff
@@ -88,9 +86,15 @@ def tr_without_district
     {
       name: 1,
       name__mn: 2,
-      party: 4,
+      party: 4
```
Comment (chrismytton) [inline scraper.rb]: This change also seems unrelated to this pull request (and is inconsistent with `tr_with_district`, which has a trailing comma on the `Hash` literal). I think it's generally best to be consistent with what's in the rest of the file.

**Review by chrismytton — CHANGES_REQUESTED** [review]

File: scraper.rb (hunk 2 above)
Comment (ondenman) [reply scraper.rb]: @current_constituency holds the value of the last district found. If the row doesn't have a district cell, @current_constituency is used. So instead of returning nothing, the methods returns @current_constitunecy.

Comment (chrismytton) [reply scraper.rb]: Ah yes, I understand now! In that case I guess you don't need the `attr_accessor :current_constituency` bit? Instead you can just return `@current_constituency` at the end of the method.

Comment (ondenman) [reply scraper.rb]: I've applied the change in a fixup. fc91033

**Review by chrismytton — APPROVED** [review]
Woop woop! ✨  🚀  ✨

File: scraper.rb (hunk 2 above)
Comment (tmtmtmtm) [reply scraper.rb]: I'm afraid I'm slower than @chrismytton and still don't understand this logic…

Comment (tmtmtmtm) [reply scraper.rb]: Ah, I think I've got it now — it's because you're treating each row as variable data, some of which hold memory from previous ones. Erk. This implementation is quite scary! In general you should think of object state as being immutable. If you find yourself wanting to change it in a loop, then that's a sign that you have another object here trying to escape. Each Membership here should really be a separate object, distinct from the table.
